### PR TITLE
feat(fabric.StaticCanvas, fabric.Canvas): limit breaking changes

### PR DESCRIPTION
**Pull request by asturur**
- keep animation module optional
- avoid cancelling all animations of different canvases
- move cancelling to static canvas
- restore chanability for simple effects

### CODE REVIEW

**Comment by asturur** [comment]
@ShaMan123 have a look

File: src/static_canvas.class.js [hunk 1]
```diff
@@ -1746,6 +1762,10 @@
       }
       this.forEachObject(function(object) {
         object.dispose && object.dispose();
+        // animation module is still optional
+        if (fabric.runningAnimations) {
+          fabric.runningAnimations.cancelByTarget(object);
```
Comment (ShaMan123) [inline src/static_canvas.class.js:1767]: not sure this is good enough because what if there's a nested object being animated under a group?

Comment (ShaMan123) [reply src/static_canvas.class.js:1767]: Give a few minutes I'm writing a function for canvas

Comment (ShaMan123) [reply src/static_canvas.class.js:1767]: redundant because it's covered by `object.dispose`

Comment (ShaMan123) [reply src/static_canvas.class.js:1767]: see #7629

Comment (asturur) [reply src/static_canvas.class.js:1767]: oh ok i did not think of object.dispose right.

Comment (asturur) [reply src/static_canvas.class.js:1767]: i want to puke, but if someone finds a good idea to animate an object clippath, when object is disposed that needs to be cancelled to. i ll add that.

Comment (asturur) [reply src/static_canvas.class.js:1767]: no maybe the clippath is an edge case to explain. But if the clippath is shared across objects, its animation should be cacnceled manually.

Comment (ShaMan123) [reply src/static_canvas.class.js:1767]: > i want to puke, but if someone finds a good idea to animate an object clippath, when object is disposed that needs to be cancelled to. i ll add that.

Funny stuff.

But I saw you merged with this line. It should be removed.

**Comment by asturur** [comment]
did you not applied a commit that was deleting it?

File: src/static_canvas.class.js (hunk 1 above)
Comment (asturur) [reply src/static_canvas.class.js:1767]: oh ok your commit was adding the code to make this line useless.
But wasn't removing this line.
I'll do that

**Comment by ShaMan123** [comment]
didn't.
opened PR #7631

**Pull request merged**